bottles/frontend/views/new.py

```diff
@@ -176,80 +162,119 @@ def set_path(_dialog, response):
         dialog.connect("response", set_path)
         dialog.show()
 
-    def create_bottle(self, *_args):
+    def create_bottle(self, *_args) -> None:
+        """ Starts creating the bottle. """
         # set widgets states
+        self.is_closable = False
         self.btn_cancel.set_visible(False)
         self.btn_create.set_visible(False)
-        self.page_create.set_visible(False)
-        self.title.set_visible(False)
+        self.set_title("")
         self.headerbar.add_css_class("flat")
-        self.stack_create.set_visible_child_name("page_creating")
         self.shortcut_escape.set_action(None)
-        
-        # avoid giant/empty window
-        self.set_default_size(450, 430)
+        self.stack_create.set_visible_child_name("page_statuses")
+        self.status_statuses.set_title(_("Creating Bottle…"))
+        self.status_statuses.set_description(_("This could take a while."))
 
-        sandbox_state = self.switch_sandbox.get_state()
-        if self.selected_env == "custom":
+        if self.check_custom.get_active():
             self.runner = self.manager.runners_available[self.combo_runner.get_selected()]
 
-        if self.combo_arch.get_selected() == 0:
-            arch = "win64"
-        else:
-            arch = "win32"
-
         RunAsync(
             task_func=self.manager.create_bottle,
             callback=self.finish,
             name=self.entry_name.get_text(),
             path=self.custom_path,
-            environment=self.selected_env,
+            environment=self.__radio_get_active(),
             runner=self.runner,
```

Comment:
Can't this be `None` now? If it is because of the environment, will it be managed correctly in manager.py?

Reply:
> will it be managed correctly in manager.py?

Yes, because manager.py automatically returns ordered runners by priority-version. I have 15 runners, picked a few random ones and they all correctly used the runner.